core/templates/pages/creator-dashboard-page/creator-dashboard-page.component.spec.ts

```diff
@@ -360,38 +367,52 @@ describe('Creator dashboard controller', () => {
         expect(ctrl.myExplorationsView).toBe('a');
       });
 
-      it('should set explorations sorting options', function() {
+      it('should reverse the sort order of explorations when the creator' +
+        ' re-selects the current sorting type', function() {
         expect(ctrl.isCurrentSortDescending).toBe(true);
         expect(ctrl.currentSortType).toBe('numOpenThreads');
         ctrl.setExplorationsSortingOptions('numOpenThreads');
         expect(ctrl.isCurrentSortDescending).toBe(false);
+      });
 
+      it('should update the exploration sort order based on the' +
+        ' option chosen by the creator', function() {
         ctrl.setExplorationsSortingOptions('new_open');
         expect(ctrl.currentSortType).toBe('new_open');
       });
 
-      it('should set subscription sorting options', function() {
+      it('should reverse the sort order of subscriptions when the creator' +
+        ' re-selects the current sorting type', function() {
         expect(ctrl.isCurrentSubscriptionSortDescending).toBe(true);
         expect(ctrl.currentSubscribersSortType).toBe('username');
         ctrl.setSubscriptionSortingOptions('username');
         expect(ctrl.isCurrentSubscriptionSortDescending).toBe(false);
+      });
 
+      it('should update the subscription sort order based on the' +
+        ' option chosen by the creator', function() {
         ctrl.setSubscriptionSortingOptions('new_subscriber');
         expect(ctrl.currentSubscribersSortType).toBe('new_subscriber');
       });
 
-      it('should sort subscription function', function() {
+      it('should sort subscription list given entity type', function() {
         var entity = {
           username: 'username'
         };
+        expect(ctrl.currentSubscribersSortType).toBe('username');
         expect(ctrl.sortSubscriptionFunction(entity)).toBe(
           'username');
-
-        ctrl.setSubscriptionSortingOptions('impact');
-        expect(ctrl.sortSubscriptionFunction({})).toBe(0);
       });
 
-      it('should sort by function', function() {
+      it('should not sort subscription list by impact given empty entity type',
```

Comment:
I am confused by this name, can you explain?

Reply:
When providing an empty object to sort the subscription list by impact, the sort should not happen.